toolchain/internal/common.bzl

```
@@ -93,9 +93,14 @@ def os_version_arch(rctx):
     _os = os(rctx)
     _arch = arch(rctx)
 
-    if _os == "linux" and not rctx.attr.exec_os:
-        (distname, version) = _linux_dist(rctx)
-        return distname, version, _arch
+    if _os == "linux":
+        if (rctx.attr.exec_linux_distribution == "") != (rctx.attr.exec_linux_distribution_version == ""):
+            fail("Either both or neither of linux_distribution and linux_distribution_version must be set")
+        if rctx.attr.exec_linux_distribution and rctx.attr.exec_linux_distribution_version:
+            return rctx.attr.exec_linux_distribution, rctx.attr.exec_linux_distribution_version, _arch
+        if not rctx.attr.exec_os:
+            (distname, version) = _linux_dist(rctx)
+            return distname, version, _arch
```

To read `/etc/os-release` when `exec_os = "linux"` + the host platform is Linux (not just when `exec_os = None`) I think we'd want:

```suggestion
        if rctx.os.name == "linux":
            (distname, version) = _linux_dist(rctx)
            return distname, version, _arch
```

---

I don't feel strongly about whether we should do the above or change the docs on `exec_linux_distribution` to describe the existing behavior but we should probably do one or the other.